Sources/Recycling/ListViews/Wishlist/Cell/WishListViewCell.swift

```diff
@@ -0,0 +1,221 @@
+            
+            rightImageDetail.trailingAnchor.constraint(equalTo: adImageView.trailingAnchor, constant: -.mediumLargeSpacing),
+            rightImageDetail.bottomAnchor.constraint(equalTo: adImageView.bottomAnchor, constant: -.mediumSpacing),
+            rightImageDetail.heightAnchor.constraint(equalTo: adImageView.heightAnchor, multiplier: 0.10),
+            rightImageDetail.widthAnchor.constraint(equalTo: adImageView.widthAnchor, multiplier: 0.20),
+            
+            subtitleContainer.leadingAnchor.constraint(equalTo: adImageView.leadingAnchor),
+            subtitleContainer.trailingAnchor.constraint(equalTo: adImageView.trailingAnchor),
+            subtitleContainer.topAnchor.constraint(equalTo: adImageView.bottomAnchor, constant: .mediumSpacing),
+            
+            titleLabel.leadingAnchor.constraint(equalTo: subtitleContainer.leadingAnchor),
+            titleLabel.trailingAnchor.constraint(equalTo: subtitleContainer.trailingAnchor),
+            titleLabel.topAnchor.constraint(equalTo: subtitleContainer.bottomAnchor),
+        ])
+    }
+    
+    public override func layoutSubviews() {
+        let roundedCornerLayer = CAShapeLayer()
+        roundedCornerLayer.bounds = rightImageDetail.frame
+        roundedCornerLayer.position = rightImageDetail.center
+        roundedCornerLayer.path = UIBezierPath(roundedRect: rightImageDetail.bounds, cornerRadius: WishListViewCell.cornerRadius).cgPath
+        rightImageDetail.layer.mask = roundedCornerLayer
+        
+        let roundedRightCornerLayer = CAShapeLayer()
+        roundedRightCornerLayer.bounds = leftImageDetail.frame
+        roundedRightCornerLayer.position = leftImageDetail.center
+        roundedRightCornerLayer.path = UIBezierPath(roundedRect: leftImageDetail.bounds, byRoundingCorners: [UIRectCorner.topRight],
+                                                    cornerRadii: CGSize(width: WishListViewCell.cornerRadius,
+                                                                        height: WishListViewCell.cornerRadius)).cgPath
+        leftImageDetail.layer.mask = roundedRightCornerLayer
+    }
+    
+    // MARK: - Superclass Overrides
+    
+    override public func prepareForReuse() {
+        adImageView.image = nil
+        leftImageDetail.text = ""
+        rightImageDetail.text = ""
+        leftSubtitleDetail.text = ""
+        rightSubtitleDetail.text = ""
+        titleLabel.text = ""
+        
+        if let model = model {
+            dataSource?.wishListViewCell(self, cancelLoadingImageForModel: model, imageWidth: adImageView.frame.size.width)
+        }
+    }
+    
+    // MARK: - Depedency injection
+    
+    /// The model contains data used to populate the view.
+    public var model: WishListViewModel? {
+        didSet {
+            guard let model = model else { return }
+            leftImageDetail.text = model.leftImageDetail
```

Comment:
I would do remove `guard let model = model else { return }`

and do

```swift
leftImageDetail.text = model?.leftImageDetail
```

instead, otherwise, if you send `cell.model = nil` it won't clear the values in the cell.